src/parseTools.js

```diff
@@ -34,33 +37,37 @@ function preprocess(text, filenameHint) {
       if (line[line.length-1] == '\r') {
         line = line.substr(0, line.length-1); // Windows will have '\r' left over from splitting over '\r\n'
       }
-      if (!line[0] || line[0] != '#') {
-        if (showStack.indexOf(false) == -1) {
-          ret += line + '\n';
+      if (isHtml && line.indexOf('<style') != -1 && !inStyle) {
+        inStyle = true;
+      }
+      if (isHtml && line.indexOf('</style') != -1 && inStyle) {
+        inStyle = false;
+      }
+
+      if (!inStyle && line.indexOf('#if') === 0) {
```

Comment:
instead of checking `!inStyle` multiple times, how about putting the whole section irrelevant to style in
```
if (!inStyle) {
  ..
}
```

Reply:
It complicates things slightly as it means I now need two 'else' blocks instead of one ... but by all means have a look and see which you prefer...